aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -379,21 +379,23 @@ public void initialize(Map<String, String> props) {
       }
     }
 
+    initMetrics(properties);
+  }
+
+  @SuppressWarnings("CatchBlockLogException")
+  private void initMetrics(Map<String, String> props) {
     // Report Hadoop metrics if Hadoop is available
     try {
       DynConstructors.Ctor<MetricsContext> ctor =
           DynConstructors.builder(MetricsContext.class)
-              .loader(S3FileIO.class.getClassLoader())
               .hiddenImpl(DEFAULT_METRICS_IMPL, String.class)
               .buildChecked();
       MetricsContext context = ctor.newInstance("s3");
-      context.initialize(properties);
+      context.initialize(props);
       this.metrics = context;
     } catch (NoClassDefFoundError | NoSuchMethodException | ClassCastException e) {
       LOG.warn(
-          "Unable to load metrics class: '{}', falling back to null metrics",
-          DEFAULT_METRICS_IMPL,
-          e);
+          "Unable to load metrics class: '{}', falling back to null metrics", DEFAULT_METRICS_IMPL);
```

Comment:
Does it make sense to make it error, so that user may know it's not working and it's using a default. This is in case some corrective action needs to be taken.

Just a thought, wdyt?

Reply:
I feel warn is an appropriate level. Currently using the Java library standalone without Hadoop you will always get this message so I think error is a little strong. (I’d actually prefer it at info).